### ISSUE TITLE
RPM package not signed

### ISSUE THREAD

**Issue by alxpl** [NordSecurity/nordvpn-linux#683]
Hello,

The latest version of the NordVPN client rpm has not been signed with your key and cannot be installed (unless the GPG verification is manually disabled). This is part of the output from dnf on Fedora 40:

```
Package nordvpn-3.19.1-1.x86_64.rpm is not signed
The downloaded packages were saved in cache until the next successful transaction.
You can remove cached packages by executing 'dnf clean packages'.
Error: GPG check FAILED 
```

**Comment by alxpl** [comment]
In the meantime, the x86_64 packages have started throwing a 404 error, I guess there are further issues with the repo.

**Comment by JerryPenguin** [comment]
I am also experiencing the same problem.

**Comment by fid509-gh** [comment]
I am also experiencing the same problem.
File './Packages/n/nordvpn-3.19.1-1.x86_64.rpm' not found on medium 'https://repo.nordvpn.com//yum/nordvpn/centos/x86_64'

**Comment by devzbysiu** [comment]
Hello, yes we restructured the code for building the app packages and the repositories and we missed the signing step :/, sorry for that. We are working on a fix. 

In the meantime this is what you can do to make sure the app is from us and to install it:

1. Verify the hash

 ```bash
sha256sum <path-to-rpm-file>
```

For example:
 ```bash
sha256sum /var/cache/dnf/repo.nordvpn.com_yum_nordvpn_centos_x86_64-37ccacb5bcc0924f/packages/nordvpn-3.19.1-1.x86_64.rpm
```
this command computes the hash of the downloaded package and you should get:
```
38e1013b2f38066f0eafbdb648f406efa4cc92b689dfd904eb1f71d29003246e  /var/cache/dnf/repo.nordvpn.com_yum_nordvpn_centos_x86_64-37ccacb5bcc0924f/packages/nordvpn-3.19.1-1.x86_64.rpm
```

3. Install the app with gpg check skipped:
```bash
sudo dnf install nordvpn --nogpgcheck
```

Regarding the 404 on the [repo](https://repo.nordvpn.com/yum/nordvpn/centos/x86_64/Packages/n/nordvpn-3.19.1-1.x86_64.rpm), we are working on that as well. It's unstable but it's possible to download the app if you try few times.

Sorry for the inconvenience, we'll fix it soon. Let me know if you have more questions

**Comment by devzbysiu** [comment]
The packages are signed now, so no need for a workaround above. Let me know if you spot anything

**Comment by alxpl** [comment]
Thank you, everything is working fine now.

**Comment by zobayer1** [comment]
I am facing this issue again after upgrading to Fedora 42.
```
Transaction failed: Signature verification failed.
OpenPGP check for package "nordvpn-3.20.2-1.x86_64" (/var/cache/libdnf5/nordvpn-9b57639910a75573/packages/nordvpn-3.20.2-1.x86_64.rpm) from repo "nordvpn" has failed: The repository does not have any OpenPGP keys configured.
```
Should I reinstall it with `nogpgcheck` flag?

**Comment by devzbysiu** [comment]
Hello @zobayer1, could you share more details how you installed the app? 

We tried to reproduce it on both fedora 41 and on 42 using installation script from https://support.nordvpn.com/hc/en-us/articles/20196094470929-Installing-NordVPN-on-Linux-distributions and directly via `dnf`, but can't reproduce it 🤔

**Comment by zobayer1** [comment]
@devzbysiu, I installed the CLI version of nordvpn from Fedora 41.
I used the installer script from the official download page
```
sh <(curl -sSf https://downloads.nordcdn.com/apps/linux/install.sh)
```
Then I upgraded to Fedora 42. I can try to uninstall, and reinstall. I will let you know if that fixes the problem.

**Comment by devzbysiu** [comment]
Hi @zobayer1, yeah that's a reasonable move. I think it shouldn't be needed to reinstall the app, but if this helps you that's great. We'll see what we can do with this upgrade case and why it breaks the package.

Let me know if it worked

**Comment by zobayer1** [comment]
@devzbysiu this is what I did to fix it, without reinstalling.
I removed the nordvpn repository from my repolist
```
sudo rm -f /etc/yum.repos.d/nordvpn.repo
sudo dnf -y update
```
Then, I downloaded the installer script from the official page for CLI installation. I ran the installer, which said NordVPN was already installed and had nothing to do. It also added the repository back to the repolist.

After that
```
sudo dnf -y update
```
This time there are no errors.
